include/internal/cryptlib.h

```diff
@@ -27,6 +27,7 @@
 
 #ifdef NDEBUG
 # define ossl_assert(x) ((x) != 0)
+# define ossl_is_null(x) (((x) == NULL) != 0)
```

Comment:
Errrrrr...  why not just this?

``` C
# define ossl_is_null(x) ((x) == NULL)
```

Reply:
Because clang complained in https://travis-ci.org/openssl/openssl/jobs/428394151,
see the outdated https://github.com/openssl/openssl/pull/7218#pullrequestreview-155311405:

```
ssl/packet.c:137:16: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
    if (((buf) == ((void*)0)))
```

Maybe I should use the following expresseion instead, does this look less odd? Or should I add a comment?

```C
# define ossl_is_null(x) (!((x) != NULL))
```

Reply:
(See my last fixup https://github.com/openssl/openssl/pull/7218/commits/7b679f0f38313fcaf5e5d7e1c30ef1f909e58614#diff-6e9d962dc8c30948fdf827ad471ec11dR30)

Reply:
and how about
`#define ossl_is_null(x) ossl_assert((x) == NULL)` ?

Reply:
Yes that looks much better. Only the logical condition needs to be inverted (twice), otherwise the assert will fail very often :wink:.
```C
#define ossl_is_null(x) (!ossl_assert((x) != NULL))
```
 But don't worry @bernd-edlinger @levitte, you two (and me) are in good company, see @mattcaswell's glitch in https://github.com/openssl/openssl/issues/7130#issuecomment-421091518. This incident again proves my conviction, that - in contrary to its silicon counterpart - the human brain is lousy at doing boolean algebra.